clinica/iotools/converters/ukb_to_bids/ukb_utils.py

```diff
@@ -0,0 +1,311 @@
+from os import PathLike
+from typing import BinaryIO, Iterable, List, Tuple, Union
+
+from pandas import DataFrame
+
+
+def find_clinical_data(
+    clinical_data_directory: PathLike,
+) -> Tuple[DataFrame, DataFrame, DataFrame, DataFrame, DataFrame]:
+    from pathlib import Path
+
+    import pandas as pd
+
+    from clinica.utils.stream import cprint
+
+    # Read the xls
+    try:
+        image_data_file = list(Path(clinical_data_directory).glob("*.csv"))
+    except StopIteration:
+        raise FileNotFoundError("Imaging collection file not found.")
+
+    # Assign the dfs
+    df_list = []
+    for i in range(0, len(image_data_file)):
+        df_list.append(pd.read_csv(str(image_data_file[i])))
+        df_list[i] = df_list[i].set_index(df_list[i].axes[1][0])
+        if df_list[i].index.name == "eid":
+            df_clinical = df_list[i]
+
+    # Warn if a dataframe is missing
+    if "df_clinical" in locals():
+        cprint(msg="All clinical data have been found", lvl="info")
+    else:
+        raise FileNotFoundError("Clinical data not found or incomplete")
+
+    return df_clinical
+
+
+def read_clinical_data(
+    clinical_data_directory: PathLike,
+) -> Tuple[DataFrame, DataFrame, DataFrame, DataFrame, DataFrame]:
+    df_clinical = find_clinical_data(clinical_data_directory)
+    dictionnary = {
+        "clinical": df_clinical,
+    }
+    return dictionnary
+
+
+def read_imaging_data(imaging_data_directory: PathLike) -> DataFrame:
+    from pathlib import Path
+
+    import pandas as pd
+
+    source_path_series = pd.Series(
+        find_imaging_data(imaging_data_directory), name="source_path"
+    )
+
+    source_dir_series = source_path_series.apply(lambda x: Path(str(x)).parent).rename(
+        "source_dir"
+    )
+
+    subject_id = source_dir_series.apply(lambda x: Path(str(x)).stem).rename(
+        "subject_id"
+    )
+
+    df_source = pd.concat(
+        [
+            source_path_series,
+            source_dir_series,
+            subject_id,
+        ],
+        axis=1,
+    )
+    # df_source = df_source.rename(
+    #     {0: "Subject", 1: "modality", 2: "Date"}, axis="columns"
+    # ).drop_duplicates()
+
+    # df_source = df_source.assign(participant_id=lambda df: "sub-" + df.Subject)
+    return df_source
+
+
+def find_imaging_data(path_to_source_data: PathLike) -> Iterable[PathLike]:
+    from pathlib import Path
+
+    for path in Path(path_to_source_data).rglob("*.nii.gz"):
+        yield str(path.relative_to(path_to_source_data))
+
+
+def intersect_data(df_source: DataFrame, dict_df: dict) -> Tuple[DataFrame, DataFrame]:
+    import pandas as pd
```

Comment:
The name of the function is not very informative, can we add a docstring to explain what we are trying to achieve with it?

Reply:
No problem ! I added a docstring, tell me if you feel it is not enough :)